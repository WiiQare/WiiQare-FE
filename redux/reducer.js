import { createSlice } from '@reduxjs/toolkit';

const initialState = {
  client: { register: {}, patient: {} },
};

export const ReducerSlice = createSlice({
  name: 'wiiqare',
  initialState,
  reducers: {
    setRegister: (state, action) => {
      state.client.register = { ...action.payload };
    },
    setPatientDispatch: (state, action) => {
      state.client.patient = { ...action.payload };
    },
  },
});

export const { setRegister, setPatientDispatch } = ReducerSlice.actions;

export default ReducerSlice.reducer;
