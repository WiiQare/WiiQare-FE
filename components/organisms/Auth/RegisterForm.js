import React from "react";
import { createContext, useState } from "react";

import { StepLabel, Stack } from "@mui/material";
import Stepper from "@mui/material/Stepper";
import Step from "@mui/material/Step";
import StepConnector, {stepConnectorClasses} from "@mui/material/StepConnector";
import { styled } from "@mui/material/styles";

import MenuHolder from "../../atoms/MenuHolder";
import BlinkSnackbar from "../../atoms/BlinkSnackbar";
import StepRegistration from "./StepRegistration";

export const FormContextRegister = createContext();

const steps = [
	"Votre adresse email",
	"VÃ©rification de l'email",
	"Information personnel",
];

const ColorlibConnector = styled(StepConnector)(({ theme }) => ({
	[`&.${stepConnectorClasses.alternativeLabel}`]: {
		top: 11,
		left: "calc(-50% + 12px)",
		right: "calc(50% + 12px)",
	},
	[`&.${stepConnectorClasses.active}`]: {
		[`& .${stepConnectorClasses.line}`]: {
			backgroundColor: theme.palette.primary.main,
		},
	},
	[`&.${stepConnectorClasses.completed}`]: {
		[`& .${stepConnectorClasses.line}`]: {
			backgroundColor: theme.palette.primary.main,
		},
	},
	[`& .${stepConnectorClasses.line}`]: {
		height: 4,
		border: 0,
		backgroundColor:
			theme.palette.mode === "dark" ? theme.palette.grey[800] : "#f0f4fd",
		borderRadius: 1,
	},
}));

const RegisterForm = () => {
	const [activeStep, setActiveStep] = React.useState(0);
	const [completed, setCompleted] = React.useState({});
    const [formData, setFormData] = useState({});

	const totalSteps = () => {
		return steps.length;
	};

	const completedSteps = () => {
		return Object.keys(completed).length;
	};

	const isLastStep = () => {
		return activeStep === totalSteps() - 1;
	};

	const allStepsCompleted = () => {
		return completedSteps() === totalSteps();
	};

	const handleNext = () => {
		const newActiveStep =
			isLastStep() && !allStepsCompleted()
				? // It's the last step, but not all steps have been completed,
				// find the first step that has been completed
				steps.findIndex((step, i) => !(i in completed))
				: activeStep + 1;
		setActiveStep(newActiveStep);
	};

	const handleBack = () => {
		setActiveStep((prevActiveStep) => prevActiveStep - 1);
	};

	const handleStep = (step) => () => {
		setActiveStep(step);
	};

	const handleComplete = (values) => {
		const newCompleted = completed;
		newCompleted[activeStep] = true;
		setCompleted(newCompleted);
		handleNext();
	};

	return (
		<>
			<div className="form-holder">
				<MenuHolder href="/login" label="S'IDENTIFIER" />
				<div className="signin-signup-form">
					<Stack
						sx={{ mb: 2, mt: 2, width: "100%" }}
						justifyContent="center"
						alignItems="center"
					>
						<Stepper
							sx={{ maxWidth: 450, width: "100%" }}
							alternativeLabel
							activeStep={activeStep}
							connector={<ColorlibConnector />}
						>
							{steps.map((label, index) => (
								<Step key={label} completed={completed[index]}>
									<StepLabel color="inherit">
										{label}
									</StepLabel>
								</Step>
							))}
						</Stepper>
					</Stack>

					<div className="form-items">

						<FormContextRegister.Provider value={{ activeStep, setActiveStep, handleComplete, formData, setFormData }}>
							<StepRegistration />
						</FormContextRegister.Provider>

					</div>
				</div>
			</div>
			{/* <BlinkSnackbar /> */}
		</>
	);
}

export default RegisterForm;
