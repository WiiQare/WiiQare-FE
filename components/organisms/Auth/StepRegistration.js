import React, { useContext } from "react";
import Email from "./Forms/email";
import Information from "./Forms/informations";
import Otp from "./Forms/otp";
import { FormContextRegister } from "./RegisterForm";

function StepRegistration() {
  const { activeStep } = useContext(FormContextRegister);
  let stepContent;

  switch (activeStep) {
    case 0:
      stepContent = <Email />;
      break;
    case 1:
      stepContent = <Otp />;
      break;
    case 2:
      stepContent = <Information />;
      break;
    default:
      break;
  }

  return stepContent;
}

export default StepRegistration;
