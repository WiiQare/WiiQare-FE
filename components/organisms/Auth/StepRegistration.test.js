import React from 'react';
import { render } from '@testing-library/react';
import StepRegistration from './StepRegistration';
import { FormContextRegister } from './RegisterForm';
import { Provider } from 'react-redux';
import { store } from '../../../redux/store';
import { QueryClientProvider, QueryClient } from 'react-query';

jest.mock('next/router', () => ({
  useRouter: jest.fn().mockReturnValue({
    router: [],
  }),
}));

describe('StepRegistration', () => {
  it('should render email if activeStep=0', () => {
    const queryClient = new QueryClient();
    const { container } = render(
      <QueryClientProvider client={queryClient}>
        <Provider store={store}>
          <FormContextRegister.Provider
            value={{
              activeStep: 0,
              setActiveStep: jest.fn(),
              handleComplete: jest.fn(),
              formData: {},
              setFormData: jest.fn(),
            }}
          >
            <StepRegistration />
          </FormContextRegister.Provider>
        </Provider>
      </QueryClientProvider>,
    );
    expect(container).toMatchSnapshot('email');
  });

  it('should render otp if activeStep=1', () => {
    const queryClient = new QueryClient();
    const { container } = render(
      <QueryClientProvider client={queryClient}>
        <Provider store={store}>
          <FormContextRegister.Provider
            value={{
              activeStep: 1,
              setActiveStep: jest.fn(),
              handleComplete: jest.fn(),
              formData: {},
              setFormData: jest.fn(),
            }}
          >
            <StepRegistration />
          </FormContextRegister.Provider>
        </Provider>
      </QueryClientProvider>,
    );
    expect(container).toMatchSnapshot('otp');
  });

  it('should render information if activeStep=2', () => {
    const queryClient = new QueryClient();
    const { container } = render(
      <QueryClientProvider client={queryClient}>
        <Provider store={store}>
          <FormContextRegister.Provider
            value={{
              activeStep: 2,
              setActiveStep: jest.fn(),
              handleComplete: jest.fn(),
              formData: {},
              setFormData: jest.fn(),
            }}
          >
            <StepRegistration />
          </FormContextRegister.Provider>
        </Provider>
      </QueryClientProvider>,
    );
    expect(container).toMatchSnapshot('information');
  });
});
