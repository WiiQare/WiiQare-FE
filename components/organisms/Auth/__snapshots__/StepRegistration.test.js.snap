// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`StepRegistration should render email if activeStep=0: email 1`] = `
<div>
  <div
    class="MuiBox-root css-1vdpy2o"
  >
    <p
      class="MuiTypography-root MuiTypography-body1 css-vkc4v8-MuiTypography-root"
    >
      Entrez votre email
    </p>
  </div>
  <form
    id="signupform"
  >
    <div
      class="css-1p5q5e5-MuiStack-root"
    >
      <div
        class="space-y-1"
      >
        <div
          class="MuiFormControl-root MuiFormControl-fullWidth MuiTextField-root lowercase css-wb57ya-MuiFormControl-root-MuiTextField-root"
        >
          <label
            class="MuiFormLabel-root MuiInputLabel-root MuiInputLabel-formControl MuiInputLabel-animated MuiInputLabel-outlined MuiFormLabel-colorPrimary MuiInputLabel-root MuiInputLabel-formControl MuiInputLabel-animated MuiInputLabel-outlined css-14s5rfu-MuiFormLabel-root-MuiInputLabel-root"
            data-shrink="false"
            for="outlined-basic"
            id="outlined-basic-label"
          >
            E-mail Address
          </label>
          <div
            class="MuiInputBase-root MuiOutlinedInput-root MuiInputBase-colorPrimary MuiInputBase-fullWidth MuiInputBase-formControl css-md26zr-MuiInputBase-root-MuiOutlinedInput-root"
          >
            <input
              aria-invalid="false"
              class="MuiInputBase-input MuiOutlinedInput-input css-1t8l2tu-MuiInputBase-input-MuiOutlinedInput-input"
              id="outlined-basic"
              name="email"
              type="text"
              value=""
            />
            <fieldset
              aria-hidden="true"
              class="MuiOutlinedInput-notchedOutline css-1d3z3hw-MuiOutlinedInput-notchedOutline"
            >
              <legend
                class="css-yjsfm1"
              >
                <span>
                  E-mail Address
                </span>
              </legend>
            </fieldset>
          </div>
        </div>
      </div>
      <div
        class="form-button"
      >
        <button
          class="MuiButtonBase-root MuiButton-root MuiButton-contained MuiButton-containedPrimary MuiButton-sizeLarge MuiButton-containedSizeLarge MuiButton-root MuiButton-contained MuiButton-containedPrimary MuiButton-sizeLarge MuiButton-containedSizeLarge css-jh47zj-MuiButtonBase-root-MuiButton-root"
          tabindex="0"
          type="submit"
        >
          L'ÉTAPE SUIVANTE
          <span
            class="MuiTouchRipple-root css-8je8zh-MuiTouchRipple-root"
          />
        </button>
      </div>
    </div>
  </form>
</div>
`;

exports[`StepRegistration should render information if activeStep=2: information 1`] = `
<div>
  <div
    class="MuiBox-root css-1vdpy2o"
  >
    <p
      class="MuiTypography-root MuiTypography-body1 css-vkc4v8-MuiTypography-root"
    >
      Information personnel
    </p>
  </div>
  <form
    id="signupform"
  >
    <div
      class="css-b55nah-MuiStack-root"
    >
      <div
        class="css-1avgqs0-MuiStack-root"
      >
        <div
          class="flex flex-col gap-1"
        >
          <div
            class="MuiFormControl-root MuiFormControl-fullWidth MuiTextField-root css-wb57ya-MuiFormControl-root-MuiTextField-root"
          >
            <label
              class="MuiFormLabel-root MuiInputLabel-root MuiInputLabel-formControl MuiInputLabel-animated MuiInputLabel-outlined MuiFormLabel-colorPrimary MuiInputLabel-root MuiInputLabel-formControl MuiInputLabel-animated MuiInputLabel-outlined css-14s5rfu-MuiFormLabel-root-MuiInputLabel-root"
              data-shrink="false"
              for="outlined-basic"
              id="outlined-basic-label"
            >
              Entrez votre nom
            </label>
            <div
              class="MuiInputBase-root MuiOutlinedInput-root MuiInputBase-colorPrimary MuiInputBase-fullWidth MuiInputBase-formControl css-md26zr-MuiInputBase-root-MuiOutlinedInput-root"
            >
              <input
                aria-invalid="false"
                class="MuiInputBase-input MuiOutlinedInput-input css-1t8l2tu-MuiInputBase-input-MuiOutlinedInput-input"
                id="outlined-basic"
                name="firstName"
                type="text"
                value=""
              />
              <fieldset
                aria-hidden="true"
                class="MuiOutlinedInput-notchedOutline css-1d3z3hw-MuiOutlinedInput-notchedOutline"
              >
                <legend
                  class="css-yjsfm1"
                >
                  <span>
                    Entrez votre nom
                  </span>
                </legend>
              </fieldset>
            </div>
          </div>
        </div>
        <div
          class="flex flex-col gap-1"
        >
          <div
            class="MuiFormControl-root MuiFormControl-fullWidth MuiTextField-root css-wb57ya-MuiFormControl-root-MuiTextField-root"
          >
            <label
              class="MuiFormLabel-root MuiInputLabel-root MuiInputLabel-formControl MuiInputLabel-animated MuiInputLabel-outlined MuiFormLabel-colorPrimary MuiInputLabel-root MuiInputLabel-formControl MuiInputLabel-animated MuiInputLabel-outlined css-14s5rfu-MuiFormLabel-root-MuiInputLabel-root"
              data-shrink="false"
              for="outlined-basic"
              id="outlined-basic-label"
            >
              Entrez votre après nom
            </label>
            <div
              class="MuiInputBase-root MuiOutlinedInput-root MuiInputBase-colorPrimary MuiInputBase-fullWidth MuiInputBase-formControl css-md26zr-MuiInputBase-root-MuiOutlinedInput-root"
            >
              <input
                aria-invalid="false"
                class="MuiInputBase-input MuiOutlinedInput-input css-1t8l2tu-MuiInputBase-input-MuiOutlinedInput-input"
                id="outlined-basic"
                name="lastName"
                type="text"
                value=""
              />
              <fieldset
                aria-hidden="true"
                class="MuiOutlinedInput-notchedOutline css-1d3z3hw-MuiOutlinedInput-notchedOutline"
              >
                <legend
                  class="css-yjsfm1"
                >
                  <span>
                    Entrez votre après nom
                  </span>
                </legend>
              </fieldset>
            </div>
          </div>
        </div>
      </div>
      <div
        class="flex flex-col gap-1"
      >
        <div
          class="MuiFormControl-root MuiFormControl-fullWidth MuiTextField-root css-wb57ya-MuiFormControl-root-MuiTextField-root"
        >
          <label
            class="MuiFormLabel-root MuiInputLabel-root MuiInputLabel-formControl MuiInputLabel-animated MuiInputLabel-shrink MuiInputLabel-outlined MuiFormLabel-colorPrimary MuiFormLabel-filled MuiInputLabel-root MuiInputLabel-formControl MuiInputLabel-animated MuiInputLabel-shrink MuiInputLabel-outlined css-1sumxir-MuiFormLabel-root-MuiInputLabel-root"
            data-shrink="true"
            for=":r3:"
            id=":r3:-label"
          >
            Numéro de téléphone
          </label>
          <div
            class="MuiInputBase-root MuiOutlinedInput-root MuiInputBase-colorPrimary MuiInputBase-fullWidth MuiInputBase-formControl MuiInputBase-adornedStart css-8j6b76-MuiInputBase-root-MuiOutlinedInput-root"
          >
            <div
              class="MuiInputAdornment-root MuiInputAdornment-positionStart MuiInputAdornment-outlined MuiInputAdornment-sizeMedium MuiPhoneNumber-positionStart css-ittuaa-MuiInputAdornment-root"
            >
              <span
                class="notranslate"
              >
                ​
              </span>
              <button
                aria-haspopup="true"
                aria-label="Select country"
                class="MuiButtonBase-root MuiIconButton-root MuiIconButton-sizeMedium MuiPhoneNumber-flagButton css-78trlr-MuiButtonBase-root-MuiIconButton-root"
                tabindex="0"
                type="button"
              >
                <svg
                  class="margin"
                  viewBox="0 85.333 512 341.333"
                >
                  <path
                    d="M0 85.331h512v341.337H0z"
                    fill="#FFF"
                  />
                  <path
                    d="M0 85.331h170.663v341.337H0z"
                    fill="#0052B4"
                  />
                  <path
                    d="M341.337 85.331H512v341.337H341.337z"
                    fill="#D80027"
                  />
                </svg>
                <span
                  class="MuiTouchRipple-root css-8je8zh-MuiTouchRipple-root"
                />
              </button>
            </div>
            <input
              aria-invalid="false"
              class="MuiInputBase-input MuiOutlinedInput-input MuiInputBase-inputAdornedStart css-1o9s3wi-MuiInputBase-input-MuiOutlinedInput-input"
              id=":r3:"
              name="phoneNumber"
              placeholder="+1 (702) 123-4567"
              type="tel"
              value="+33"
            />
            <fieldset
              aria-hidden="true"
              class="MuiOutlinedInput-notchedOutline css-1d3z3hw-MuiOutlinedInput-notchedOutline"
            >
              <legend
                class="css-14lo706"
              >
                <span>
                  Numéro de téléphone
                </span>
              </legend>
            </fieldset>
          </div>
        </div>
      </div>
      <div
        class="flex flex-col gap-1"
      >
        <div
          class="MuiFormControl-root MuiFormControl-fullWidth css-q8hpuo-MuiFormControl-root"
        >
          <label
            class="MuiFormLabel-root MuiInputLabel-root MuiInputLabel-formControl MuiInputLabel-animated MuiInputLabel-outlined MuiFormLabel-colorPrimary MuiInputLabel-root MuiInputLabel-formControl MuiInputLabel-animated MuiInputLabel-outlined css-14s5rfu-MuiFormLabel-root-MuiInputLabel-root"
            data-shrink="false"
            for="outlined-basic1"
          >
            Mot de passe
          </label>
          <div
            class="MuiInputBase-root MuiOutlinedInput-root MuiInputBase-colorPrimary MuiInputBase-formControl MuiInputBase-adornedEnd css-o9k5xi-MuiInputBase-root-MuiOutlinedInput-root"
          >
            <input
              aria-invalid="false"
              class="MuiInputBase-input MuiOutlinedInput-input MuiInputBase-inputAdornedEnd css-nxo287-MuiInputBase-input-MuiOutlinedInput-input"
              id="outlined-basic1"
              name="password"
              type="password"
              value=""
            />
            <div
              class="MuiInputAdornment-root MuiInputAdornment-positionEnd MuiInputAdornment-outlined MuiInputAdornment-sizeMedium css-1laqsz7-MuiInputAdornment-root"
            >
              <button
                aria-label="toggle password visibility"
                class="MuiButtonBase-root MuiIconButton-root MuiIconButton-edgeEnd MuiIconButton-sizeMedium css-1yq5fb3-MuiButtonBase-root-MuiIconButton-root"
                tabindex="0"
                type="button"
              >
                <svg
                  aria-hidden="true"
                  class="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium css-i4bv87-MuiSvgIcon-root"
                  data-testid="VisibilityIcon"
                  focusable="false"
                  viewBox="0 0 24 24"
                >
                  <path
                    d="M12 4.5C7 4.5 2.73 7.61 1 12c1.73 4.39 6 7.5 11 7.5s9.27-3.11 11-7.5c-1.73-4.39-6-7.5-11-7.5zM12 17c-2.76 0-5-2.24-5-5s2.24-5 5-5 5 2.24 5 5-2.24 5-5 5zm0-8c-1.66 0-3 1.34-3 3s1.34 3 3 3 3-1.34 3-3-1.34-3-3-3z"
                  />
                </svg>
                <span
                  class="MuiTouchRipple-root css-8je8zh-MuiTouchRipple-root"
                />
              </button>
            </div>
            <fieldset
              aria-hidden="true"
              class="MuiOutlinedInput-notchedOutline css-1d3z3hw-MuiOutlinedInput-notchedOutline"
            >
              <legend
                class="css-yjsfm1"
              >
                <span>
                  Mot de passe
                </span>
              </legend>
            </fieldset>
          </div>
        </div>
      </div>
      <div
        class="flex flex-col gap-1"
      >
        <div
          class="MuiFormControl-root MuiFormControl-fullWidth css-q8hpuo-MuiFormControl-root"
        >
          <label
            class="MuiFormLabel-root MuiInputLabel-root MuiInputLabel-formControl MuiInputLabel-animated MuiInputLabel-outlined MuiFormLabel-colorPrimary MuiInputLabel-root MuiInputLabel-formControl MuiInputLabel-animated MuiInputLabel-outlined css-14s5rfu-MuiFormLabel-root-MuiInputLabel-root"
            data-shrink="false"
            for="outlined-basic2"
          >
            Confirmez le mot de passe
          </label>
          <div
            class="MuiInputBase-root MuiOutlinedInput-root MuiInputBase-colorPrimary MuiInputBase-formControl MuiInputBase-adornedEnd css-o9k5xi-MuiInputBase-root-MuiOutlinedInput-root"
          >
            <input
              aria-invalid="false"
              class="MuiInputBase-input MuiOutlinedInput-input MuiInputBase-inputAdornedEnd css-nxo287-MuiInputBase-input-MuiOutlinedInput-input"
              id="outlined-basic2"
              name="confirm_password"
              type="password"
              value=""
            />
            <div
              class="MuiInputAdornment-root MuiInputAdornment-positionEnd MuiInputAdornment-outlined MuiInputAdornment-sizeMedium css-1laqsz7-MuiInputAdornment-root"
            >
              <button
                aria-label="toggle password visibility"
                class="MuiButtonBase-root MuiIconButton-root MuiIconButton-edgeEnd MuiIconButton-sizeMedium css-1yq5fb3-MuiButtonBase-root-MuiIconButton-root"
                tabindex="0"
                type="button"
              >
                <svg
                  aria-hidden="true"
                  class="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium css-i4bv87-MuiSvgIcon-root"
                  data-testid="VisibilityIcon"
                  focusable="false"
                  viewBox="0 0 24 24"
                >
                  <path
                    d="M12 4.5C7 4.5 2.73 7.61 1 12c1.73 4.39 6 7.5 11 7.5s9.27-3.11 11-7.5c-1.73-4.39-6-7.5-11-7.5zM12 17c-2.76 0-5-2.24-5-5s2.24-5 5-5 5 2.24 5 5-2.24 5-5 5zm0-8c-1.66 0-3 1.34-3 3s1.34 3 3 3 3-1.34 3-3-1.34-3-3-3z"
                  />
                </svg>
                <span
                  class="MuiTouchRipple-root css-8je8zh-MuiTouchRipple-root"
                />
              </button>
            </div>
            <fieldset
              aria-hidden="true"
              class="MuiOutlinedInput-notchedOutline css-1d3z3hw-MuiOutlinedInput-notchedOutline"
            >
              <legend
                class="css-yjsfm1"
              >
                <span>
                  Confirmez le mot de passe
                </span>
              </legend>
            </fieldset>
          </div>
        </div>
      </div>
      <div
        class="flex items-center !mt-10 mb-2"
      >
        <input
          class="w-4 h-4 text-orange bg-gray-100 border-gray-300 rounded focus:ring-orange focus:ring-1"
          id="link-checkbox"
          type="checkbox"
          value=""
        />
        <label
          class="h-4 ml-2 text-sm font-normal text-gray-600 dark:text-gray-300"
          for="link-checkbox"
        >
          j'accepte les
           
          <a
            class="text-primary hover:underline"
            href="https://wiiqare.com/privacy/"
          >
            termes et conditions
          </a>
           
          d'utilisation.
        </label>
      </div>
      <div
        class="MuiBox-root css-0"
      >
        <button
          class="MuiButtonBase-root MuiButton-root MuiButton-contained MuiButton-containedPrimary MuiButton-sizeLarge MuiButton-containedSizeLarge Mui-disabled MuiButton-root MuiButton-contained MuiButton-containedPrimary MuiButton-sizeLarge MuiButton-containedSizeLarge disabled:bg-gray-200 css-jh47zj-MuiButtonBase-root-MuiButton-root"
          disabled=""
          tabindex="-1"
          type="submit"
        >
          CRÉER UN COMPTE
        </button>
      </div>
    </div>
  </form>
</div>
`;

exports[`StepRegistration should render otp if activeStep=1: otp 1`] = `
<div>
  <div
    class="MuiBox-root css-1vdpy2o"
  >
    <p
      class="MuiTypography-root MuiTypography-body1 css-vkc4v8-MuiTypography-root"
    >
      Un code vous a été envoyé, verifiez votre email
    </p>
  </div>
  <div
    class="MuiBox-root css-yzgae"
  >
    <span
      class="MuiTypography-root MuiTypography-caption css-ol7wxs-MuiTypography-root"
    >
      Expire dans
       
      <span>
        02
        :
        00
      </span>
    </span>
  </div>
  <form
    id="signupform"
  >
    <div
      class="form-text code"
    >
      <div
        style="display: flex;"
      >
        <div
          style="display: flex; align-items: center;"
        >
          <input
            aria-label="Please enter verification code. Digit 1"
            autocomplete="off"
            class="  "
            style="width: 1em; text-align: center; padding: 0px;"
            type="tel"
            value=""
          />
        </div>
        <div
          style="display: flex; align-items: center;"
        >
          <input
            aria-label="Digit 2"
            autocomplete="off"
            class=" "
            style="width: 1em; text-align: center; padding: 0px;"
            type="tel"
            value=""
          />
        </div>
        <div
          style="display: flex; align-items: center;"
        >
          <input
            aria-label="Digit 3"
            autocomplete="off"
            class=" "
            style="width: 1em; text-align: center; padding: 0px;"
            type="tel"
            value=""
          />
        </div>
        <div
          style="display: flex; align-items: center;"
        >
          <input
            aria-label="Digit 4"
            autocomplete="off"
            class=" "
            style="width: 1em; text-align: center; padding: 0px;"
            type="tel"
            value=""
          />
        </div>
        <div
          style="display: flex; align-items: center;"
        >
          <input
            aria-label="Digit 5"
            autocomplete="off"
            class=" "
            style="width: 1em; text-align: center; padding: 0px;"
            type="tel"
            value=""
          />
        </div>
        <div
          style="display: flex; align-items: center;"
        >
          <input
            aria-label="Digit 6"
            autocomplete="off"
            class=" "
            style="width: 1em; text-align: center; padding: 0px;"
            type="tel"
            value=""
          />
        </div>
      </div>
    </div>
    <div
      class="form-text text-holder"
    >
      <span
        class="text-only"
      >
        Pas réçu de code ?
         
        <button
          class="text-primary"
          type="button"
        >
          Renvoyer le code
        </button>
      </span>
    </div>
    <div
      class="form-button"
    >
      <button
        class="MuiButtonBase-root MuiButton-root MuiButton-contained MuiButton-containedPrimary MuiButton-sizeLarge MuiButton-containedSizeLarge MuiButton-root MuiButton-contained MuiButton-containedPrimary MuiButton-sizeLarge MuiButton-containedSizeLarge css-jh47zj-MuiButtonBase-root-MuiButton-root"
        tabindex="0"
        type="submit"
      >
        L'ÉTAPE SUIVANTE
        <span
          class="MuiTouchRipple-root css-8je8zh-MuiTouchRipple-root"
        />
      </button>
    </div>
  </form>
</div>
`;
